internal/core/items/component_container.rs

```diff
@@ -66,6 +69,44 @@ impl ComponentContainer {
         };
 
         let product = factory.build(factory_context);
+        if let Some(item_tree) = product.clone() {
+            let item_tree = vtable::VRc::borrow_pin(&item_tree);
+            let root_item = item_tree.as_ref().get_item_ref(0);
+            if let Some(window_item) =
+                crate::items::ItemRef::downcast_pin::<crate::items::WindowItem>(root_item)
+            {
+                // Do _not_ use a two-way binding: That causes evaluations of width and height to
+                // assert on recursive property evaluation.
```

Comment:
Is there any way to test this behavior in our tests? (i.e. that the inside is sized according to the outside)

Reply:
true, there should bet a test.

The way i'd do it is do a test would be to have a TouchArea that write its with and height to a variable.

So the "inner" would be:

``` 
export component Inner {
    out property <{x: length, y: length, width: length, height: length}> result;
    TouchArea {
       clicked => { 
          result =  { width: self.width ,  height:  self.height, x:  self.absolute-pos.x, y: self.absolute-pos.y };
       }
    }
}
```

Then the rust test function in  will issue a click where it is supposed to be.  And then query the factory (which is in a Rc) for the result property

Reply:
Yes there is already a test for recursion, but there is no test that this change is actually doing anything (i.e: setting the with and height in the inside)